### PR TITLE
Add LSQ/LSQ+ in QAT

### PR DESCRIPTION
**Pull request by RachelXu7**
### PR types
New features
### PR changes
APIs
### Describe
Add LSQ/LSQ+ in QAT 
This version adapts to the current QAT API. But some args are fixed. (symmetric is True)

### CODE REVIEW

File: python/paddle/nn/quant/lsq.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+class FakeQuantActLSQPlus(Layer):
+
+    def __init__(self,
+                 quant_bits,
+                 all_postive=False,
+                 symmetric=False,
+                 batch_init=20,
+                 dtype='float32',
+                 name=None,
+                 reduce_type=None):
```
Comment (yghstill) [inline python/paddle/nn/quant/lsq.py:125]: 增加下注释

File: python/paddle/nn/quant/lsq.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            q_a = LsqFunc.apply(activation,
+                                self.s,
+                                self.g,
+                                self.Qn,
+                                self.Qp,
+                                per_channel=False)
+        return q_a
+
+
+class FakeQuantWeightLSQPlus(Layer):
```
Comment (yghstill) [inline python/paddle/nn/quant/lsq.py:221]: same as above

File: python/paddle/fluid/contrib/slim/quantization/imperative/qat.py [hunk 3]
```diff
@@ -323,16 +323,18 @@ def __init__(self,
                 "%s is unspported to be quantized." % layer
 
         quantize_type = {
-            'abs_max', 'moving_average_abs_max', 'channel_wise_abs_max'
+            'abs_max', 'moving_average_abs_max', 'channel_wise_abs_max',
+            'lsq_weight', 'channel_wise_lsq_weight'
         }
+        act_quantize_type = {'moving_average_abs_max', 'lsq_act'}
```
Comment (yghstill) [inline python/paddle/fluid/contrib/slim/quantization/imperative/qat.py:329]: lsq_act -> lsq

File: python/paddle/fluid/contrib/slim/quantization/imperative/qat.py [hunk 4]
```diff
@@ -323,16 +323,18 @@ def __init__(self,
                 "%s is unspported to be quantized." % layer
 
         quantize_type = {
-            'abs_max', 'moving_average_abs_max', 'channel_wise_abs_max'
+            'abs_max', 'moving_average_abs_max', 'channel_wise_abs_max',
```
Comment (yghstill) [inline python/paddle/fluid/contrib/slim/quantization/imperative/qat.py:326]: 将quantize_type改成weight_quantize_type
lsq_weight -> lsq_layer_wise
channel_wise_lsq_weight -> lsq_channel_wise

**Review by yghstill — DISMISSED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_imperative_qat_lsq PROPERTIES TIMEOUT 300)`

**Review by yghstill — APPROVED** [review]

**Pull request merged**